src/fields/fields.service.ts

```
@@ -303,7 +305,114 @@ export class FieldsService {
+      // Step 4: Handle comma-separated values (current workflow)
+      if (checkboxValue.includes(',')) {
+        return checkboxValue.split(',').map((v) => v.trim());
+      }
+
+      // Step 5: Handle single value based on field type in schema
+      const fieldType = fieldMetadata.type; // Type from schema: "boolean", "string", "array"
+
+      switch (fieldType) {
+        case 'boolean':
+          const booleanResult =
+            checkboxValue === 'true' || checkboxValue === '1';
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)